packages/@sanity/form-builder/src/inputs/ObjectInput/fieldGroups/Group.tsx

```diff
@@ -0,0 +1,88 @@
+import React, {forwardRef, useEffect} from 'react'
+import {Tab} from '@sanity/ui'
+import {FieldGroup} from '@sanity/types/src'
+import {unstable_useConditionalProperty as useConditionalProperty} from '@sanity/base/hooks'
+
+interface GroupType extends FieldGroup {
+  onClick: (string) => void
+  autoFocus?: boolean
+  selected: boolean
+  onStateChange?: (name: string, hidden: boolean) => void
+  parent: unknown
+  'aria-controls': string
+  disabled?: boolean
+}
+
+export const GroupTab = forwardRef(function GroupTab(
+  props: GroupType,
+  ref: React.Ref<HTMLButtonElement>
+) {
+  const {name, title} = props
+  // Separate props for resolving conditional hidden groups
+  const {onClick, onStateChange, parent, hidden, ...group} = props
+
+  const isHidden = useConditionalProperty({
+    checkProperty: hidden,
+    checkPropertyKey: 'hidden',
+    value: group,
+    parent,
+  })
+
+  // Here goes the content of our component
+  const handleClick = React.useCallback(() => {
+    onClick(name)
+  }, [name, onClick])
+
+  useEffect(() => {
+    onStateChange(name, Boolean(isHidden))
```

Comment:
Any way we can avoid triggering a state update in the parent component as a side effect of a state update in this one? Maybe we could somehow lift the isHidden calculation up one level?

Reply:
The tricky part here is that our conditional logic is a hook and component that processes the group, and this means it can't easily be used in the parent component. We could possibly create a smarter hook that processes the array of groups instead.